src/fstar/FStar.Main.fs

```diff
@@ -211,9 +217,20 @@ let handle_error e =
 let main () =
   try
     setup_hooks ();
-    let _, time = FStar.Util.record_time go in
+    let _, time = Util.record_time go in
+    if Options.print () || Options.print_in_place () then
+      match !fstar_files with
+      | Some filenames ->
+          let printing_mode =
+            if Options.print () then
+              FStar.Prettyprint.FromTempToStdout
+            else
+              FStar.Prettyprint.FromTempToFile
+          in
+          FStar.Prettyprint.generate printing_mode filenames
```

Comment:
Doesn't this, combined with the `ToTempFile` call above, make the pretty printer run twice? Once on the input file towards the temp file, and then again from the (pretty-printed) temp file to the original file. Couldn't we just `mv` the temp file to replace the original file?

Reply:
Yes, it does. This is on purpose and perhaps I should document it somewhere. The reason is that in a some cases the layout of the original piece of code can change so much that the heuristics we use to place comments can yield slightly different placements between two runs of the prettyprinter. Since it's cheap, to me this option seems preferable to complicating the (already quite complicated) comments placements logic further.